pkg/processor/build/builder.go

```diff
@@ -403,14 +412,22 @@ func (b *Builder) writeFunctionSourceCodeToTempFile(functionSourceCode string) (
 	}
 
 	// we will generate a file named as per specified by the handler
-	moduleFileName, _, err := functionconfig.ParseHandler(b.options.FunctionConfig.Spec.Handler)
+	moduleFileName, handlerName, err := functionconfig.ParseHandler(b.options.FunctionConfig.Spec.Handler)
 	if err != nil {
 		return "", errors.Wrap(err, "Failed to parse handler")
 	}
 
-	// if the module name already an extension, leave it
-	if !strings.Contains(moduleFileName, ".") {
-		moduleFileName = fmt.Sprintf("%s.%s", moduleFileName, runtimeExtension)
+	if b.options.FunctionConfig.Spec.Runtime == "java_src" {
```

Comment:
We need to keep working hard that as little runtime specific logic seeps into the builder. Please try to think of a way to as the runtime interface something, have all runtimes return the same thing to retain previous behavior and Java return the applicable different response